### PR TITLE
Bug fix: Routes can support more than one auth type

### PR DESCRIPTION
**Pull request by jaquish**
Additionally, determine if route is noauth from the auth attribute, not host attribute.

### CODE REVIEW

File: Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBTransportBaseClient.m [hunk 1]
```diff
@@ -64,13 +64,15 @@ - (NSDictionary *)headersWithRouteInfo:(NSDictionary<NSString *, NSString *> *)r
                        byteOffsetStart:(NSNumber *)byteOffsetStart
                          byteOffsetEnd:(NSNumber *)byteOffsetEnd {
   NSString *routeStyle = routeAttributes[@"style"];
-  NSString *routeHost = routeAttributes[@"host"];
-  NSString *routeAuth = routeAttributes[@"auth"];
+
+  NSString *routeAuthStr = routeAttributes[@"auth"];
+  // routeAuths is one of user|team|app|noauth|app, user
+  NSArray<NSString *> *routeAuths = [routeAuthStr componentsSeparatedByString:@", "];
```
Comment (xuyanke) [inline Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBTransportBaseClient.m:70]: nit: use comma without whitespace as separator may be more flexible. I don't think there is anything force the white space between auth types, either "app, user" and "app,user" may be possible value of the auth attribute.

**Review by xuyanke — COMMENTED** [review]
thanks for the update. one more comment about string separator.

File: Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBTransportBaseClient.m (hunk 1 above)
Comment (jaquish) [reply Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBTransportBaseClient.m]: Updated.

**Comment by xuyanke** [comment]
lgtm.

**Pull request merged**